### ISSUE TITLE
IE11 checkboxes aren't clickable 

### ISSUE THREAD

**Issue by KalobTaulien** [wagtail/wagtail#5903]
### Issue Summary

When trying to check a checkbox in IE11 it doesn't check, and displays the hidden help text around it instead. 

### Steps to Reproduce

1. Start a new project, or open an existing project. 
2. Run `python manage.py runserver 0.0.0.0:8000` so your site is accessible to your browser
3. Open a page with checkboxes in BrowserStack or LambdaTest. This is the page I've been using: http://localhost:8000/admin/groups/2/ (but any Group Edit view will work)

I think this is a bug somewhere because checking a checkbox in any browser should be supported as there isn't too much going on behind the scenes here. 

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: **Yes**, used a fresh install with Wagtail 2.8, Python 3.7.2, and Django 2.2.11 and got the same issue. 

### Technical details

* Python version: 3.6.9
* Django version: 2.2.9
* Wagtail version: 2.8
* Browser version: www.whatsmybrowser.org/b/58AG7

Visual of the issue
![ie-checkbox-issue](https://user-images.githubusercontent.com/4743971/77257007-4da6ce80-6c37-11ea-86f6-9ea3ff237555.gif)

**Comment by thibaudcolas** [comment]
Just tried this out in Wagtail 2.10 and I got the same result. Interestingly the checkboxes in the User’s Role settings work, so this might not be an issue with all checkboxes, although I didn’t investigate further.

As a workaround it looks like navigating through the checkboxes with a keyboard and checking them works, although the focus indicator sometimes disappears.

**Comment by lb-** [comment]
@thibaudcolas  & @KalobTaulien - I have found what I think is the root cause of this
https://github.com/wagtail/wagtail/blob/master/client/scss/tools/_mixins.general.scss#L71-L78

The labels for the input checkboxes use .visuallyhidden
https://github.com/wagtail/wagtail/blob/master/wagtail/users/templates/wagtailusers/groups/includes/formatted_permissions.html#L25

And when IE triggers the click on the `input` this appears to trigger the `:active` on this label which makes it appear 'under' the mouse, and hence the click handler does not correctly work on the checkbox.

I am not sure on the best way forward as I could not see why we want the focus/active state of `.visuallyhidden` to be visible? This seems odd.

A potential quick work around could be to move the labels under the `input checkbox`, so that when the label does appear in IE, it will not take the click handler away from the input. I have tested this solution in IE 11 and it works.

https://github.com/lb-/wagtail/commit/cc0c24c77a69393f0a732d05e710c6a801e2f38a

Let me know if you wan me to make a PR with this solution.

**Comment by thibaudcolas** [comment]
Thank you @lb-, I think that’s spot on.

I assume the `:focus` and `:active` modifiers are there to make the "visually hidden" content appear again when keyboard-focused. I believe this is there with the intention that if what’s visually hidden is interactive, then you would want it to appear when focused / "active". However I’m not sure how appropriate that is – except for skip links, that behavior doesn’t feel desirable.

Your quick solution sounds good to me, but a potentially even quicker one would be to remove those `:focus` / `:active` styles from the `visuallyhidden` mixin. Apart from the labels in IE11, the only part of Wagtail this would affect is:

https://github.com/wagtail/wagtail/blob/df164a7e6938f7faf9d75efae31ee39e9ccce9c3/wagtail/snippets/templates/wagtailsnippets/snippets/type_index.html#L40

Based on the logic at https://github.com/wagtail/wagtail/blob/df164a7e6938f7faf9d75efae31ee39e9ccce9c3/wagtail/snippets/static_src/wagtailsnippets/js/snippet-multiple-select.js#L20-L30,

it looks like this really should be using the HTML `hidden` attribute so this control is hidden for screen reader users as well. And there is no value in having this being focusable and appearing on focus.
